src/error.rs

```diff
@@ -76,7 +24,57 @@ macro_rules! impl_pyerr {
     };
 }
 
-impl_pyerr!(ShapeError);
+/// Represents that dimensionalities of the given arrays don't match.
```

Comment:
I think "dimensionality" is a unnecessary. Why not just the "dimension of the given arrays" and similarly in the type name?

Reply:
`DimensionError` may be misread as "dimensions mismatch" (I would *definitely* read it as that myself) which would be incorrect because dimensions=shape and we don't check any shapes but rather only `ndim`. Another option is `NdimError` but it's rather ugly (although still an option).

E.g.: [`ndarray::ArrayBase::into_dimensionality()`](https://docs.rs/ndarray/0.15.4/ndarray/struct.ArrayBase.html#method.into_dimensionality)

Reply:
Fair enough. Let's avoid the confusion.